app/models/reservation.rb

```diff
@@ -159,6 +168,10 @@ def assign_actuals_off_reserve
     self.actual_end_at   ||= reserve_end_at
   end
 
+  def assign_category_attribute(user, params)
+    assign_attributes(category: params[:category]) if user.operator_of?(product.facility)
```

Comment:
I said in a previous note that I don't think this check will be necessary, but if we do need it, we're preferring to use cancan over explicit use of of `operator_of?`.